eng/pipelines/templates/jobs/ci.yml

```diff
@@ -393,22 +394,41 @@ jobs:
         parameters:
           JdkVersion: ${{ parameters.JavaBuildVersion }}
 
-      - task: PowerShell@2
-        displayName: Verify Code Generation
-        inputs:
-          pwsh: true
-          filePath: $(Build.SourcesDirectory)/eng/scripts/Compare-CurrentToCodegeneration.ps1
-          arguments: >
-            -Directory sdk/${{ parameters.ServiceDirectory }}
+      - ${{ if and(eq(variables['Build.Reason'], 'PullRequest'), eq(parameters.ServiceDirectory, 'auto')) }}:
+        - task: PowerShell@2
+          displayName: Verify Code Generation for Pull Request
+          inputs:
+            pwsh: true
+            filePath: $(Build.SourcesDirectory)/eng/scripts/Compare-CurrentToCodegeneration.ps1
+            arguments: >
+              -ServiceDirectories '$(PRServiceDirectories)'
+      - ${{ else }}:
+        - task: PowerShell@2
+          displayName: Verify Code Generation
+          inputs:
+            pwsh: true
+            filePath: $(Build.SourcesDirectory)/eng/scripts/Compare-CurrentToCodegeneration.ps1
+            arguments: >
+              -ServiceDirectories '${{ parameters.ServiceDirectory }}'
```

Comment:
nit: but should we put the template condition just around this one line instead of the entire step?

Reply:
Actually instead of doing the forking everywhere why not just do it once above and set PRServiceDirectories to the one service directory if it is not auto.

Reply:
Good call. I completely refactored this. The conditions are only used to set the PRServiceDirectories and everything that used PRServiceDirectories or parameters.ServiveDirectory now just uses the PRServiceDirectories variable.